crates/vm2/src/precompiles/mod.rs

```diff
@@ -82,18 +82,24 @@ impl PrecompileOutput {
 impl From<U256> for PrecompileOutput {
     fn from(value: U256) -> Self {
         Self {
-            buffer: [value, U256::zero()],
+            buffer: [value, U256::zero(), U256::zero()],
             len: 1,
             cycle_stats: None,
         }
     }
 }
 
-impl From<[U256; 2]> for PrecompileOutput {
-    fn from(value: [U256; 2]) -> Self {
+impl<const N: usize> From<[U256; N]> for PrecompileOutput
+where
+    [U256; N]: Default,
```

Comment:
This conversion is too general and doesn't make sense for `N > 3`. If you want to avoid copy-paste please define a macro via macro_rules taking the array length as a param and invoke it for 2 and 3 (and optionally, 1):

```rust
macro_rules! impl_from_array_for_precompile_output {
    ($n:tt) => {
        impl From<[U256; $n]> for PrecompileOutput {
            fn from(value: [U256; $n]) -> Self {
                todo!()
            }
        }
    };
}

impl_from_array_for_precompile_output!(2);
impl_from_array_for_precompile_output!(3);
```